platform/jsonschema/jsonschema.go

```diff
@@ -0,0 +1,89 @@
+package jsonschema
+
+import (
+	"fmt"
+	"github.com/fewlinesco/gojsonschema"
+)
+
+// AdditionalPropertyError is returned when the validation failed due to additional properties being present in the JSON input.
+type AdditionalPropertyError struct {
+	Details map[string]string
+}
+
+func (err AdditionalPropertyError) Error() string {
+	return "the json input contains unknown keys"
+}
+
+// MissingPropertyError is returned when the validation failed due to missing required properties in the JSON input.
+type MissingPropertyError struct {
+	Details map[string]string
+}
+
+func (err MissingPropertyError) Error() string {
+	return "the json input is missing required keys"
+}
+
+// ValidationError is returned when the validation failed
+type ValidationError struct {
+	Details map[string]string
+}
+
+func (err ValidationError) Error() string {
+	return "the json input is not valid against the JSON schema"
+}
+
+// ValidateJSONAgainstSchema validate some json data against a json schema and return detailed validation errors if any is present.
+func ValidateJSONAgainstSchema(jsonData []byte, jsonSchema []byte) error {
```

Comment:
a bit nitpicking but `jsonschema.ValidateJSONAgainstSchema` seems a bit redundant to me ˆˆ'